controllers/update_database_configuration.go

```diff
@@ -71,21 +74,28 @@ func (u updateDatabaseConfiguration) reconcile(ctx context.Context, r *Foundatio
 	currentConfiguration.ExcludedServers = nil
 	cluster.ClearMissingVersionFlags(&currentConfiguration)
 
+	runningVersion, err := fdbv1beta2.ParseFdbVersion(cluster.GetRunningVersion())
+	if err != nil {
+		return &requeue{curError: err, delayedRequeue: true}
+	}
+
 	if initialConfig || !equality.Semantic.DeepEqual(desiredConfiguration, currentConfiguration) {
-		var nextConfiguration fdbtypes.DatabaseConfiguration
+		var nextConfiguration fdbv1beta2.DatabaseConfiguration
 		if initialConfig {
 			nextConfiguration = desiredConfiguration
 		} else {
 			nextConfiguration = currentConfiguration.GetNextConfigurationChange(desiredConfiguration)
 		}
 		configurationString, _ := nextConfiguration.GetConfigurationString(cluster.Spec.Version)
 
-		dataState := status.Cluster.Data.State
-		if !(initialConfig || dataState.Healthy) {
-			logger.Info("Waiting for data distribution to be healthy", "stateName", dataState.Name, "stateDescription", dataState.Description)
-			r.Recorder.Event(cluster, corev1.EventTypeNormal, "NeedsConfigurationChange",
-				fmt.Sprintf("Spec require configuration change to `%s`, but data distribution is not fully healthy: %s (%s)", configurationString, dataState.Name, dataState.Description))
-			return nil
+		if !initialConfig {
```

Comment:
If the cluster was not already configured then we wouldn't want to do these checks?